exercises/build.gradle

```diff
@@ -1,44 +1,67 @@
-subprojects { project ->
-  apply plugin: "java"
-
-  sourceSets {
-    // Verify that the tests are working by running them against the example code.
-    // By replacing the "main" sourceSet with the example code we avoid any collisions
-    //   with solution code that may have been included as a "starter" (e.g. etl).
-    main {
-      java.srcDirs = ["src/example/java"]
-    }
-    project["compileJava"].doFirst { compileTask ->
-      println "  (source = ${compileTask.source.asPath})"
-    }
+// This root Gradle project is used for:
+//
+// - Checking that every test suite compiles against its corresponding example implementation;
+// - Checking that every example implementation passes its corresponding test suite;
+// - Checking that every provided starter implementation compiles.
+//
+// This file is never delivered to Exercism users.
 
-    starterSource {
-      java.srcDirs = ["src/main/java"]
-    }
-    project["compileStarterSourceJava"].doFirst { compileTask ->
-      println "  (source = ${compileTask.source.asPath})"
-    }
+def generatedTestSourceDir = "build/gen/test/java"
 
-    // In lieu of being able to disable @Ignore in JUnit tests, we filter
-    // those annotations, placing the edited tests in the path named here.
-    test {
-      java.srcDirs = ["build/gen/test/java"]
-    }
-    project["compileTestJava"].doFirst { compileTask ->
-      println "  (source = ${compileTask.source.asPath})"
-    }
-  }
+subprojects {
 
+  // Add a task that copies test source files into a build directory. All @Ignore annotations
+  // are removed during this copying, so that when we run the copied tests, none are skipped and
+  // all should execute.
+  //
   // Note that journey-test.sh also strips @Ignore annotations using sed. The
   // stripping implementations here and in journey-test.sh should be kept
   // consistent.
   task copyTestsFilteringIgnores(type: Copy) {
     from "src/test/java"
-    into "build/gen/test/java"
+    into generatedTestSourceDir
     filter { line ->
-      line.contains("@Ignore") ? "" : line
+      line.contains("@Ignore") ? null : line
```

Comment:
`null` here just skips the line, rather than including an unnecessary blank line.